src/Containers/App/index.jsx

```diff
@@ -777,33 +778,33 @@ class App extends Component {
     }
 
     try {
-      let apiVersion = await this.serial.getApiVersion().catch((e) => {
-        if (e.message.includes("Timed out after")) {
-          return null;
-        }
-        throw e;
-      });
+      let apiVersion = null;
       
-      if (apiVersion === null) {
-        let hasResets = false;
-        let i = 0;
-
-        while ((await this.serial.getFourWayInterfaceInfo(i).catch(() => null)) !== null) {
-          try {
-            await this.serial.resetFourWayInterface(i);
-            hasResets = true;
-          } catch(e) {
-            this.addLogMessage('resetEscFailed', { index: i + 1 });
+      try {
+        apiVersion = await this.serial.getApiVersion();
+      } catch(e) {
+        if (e instanceof TimeoutError) {
+          let hasResets = false;
+          let i = 0;
+
+          while ((await this.serial.getFourWayInterfaceInfo(i).catch(() => null)) !== null) {
```

Comment:
I am still not a fan of this. I think the while could also be wrapped in a try/catch and our stop condition is basically the exception being thrown.

Reply:
changed